src/components/common/ColonyActions/ActionDetailsPage/DefaultMotion/MotionPhaseWidget/StakingWidget/StakingWidgetProvider.tsx

```diff
@@ -0,0 +1,260 @@
+import React, {
+  createContext,
+  ReactNode,
+  useCallback,
+  useContext,
+  useMemo,
+  useState,
+} from 'react';
+import Decimal from 'decimal.js';
+
+import { useGetUserReputationQuery } from '~gql';
+import {
+  useAppContext,
+  useColonyContext,
+  useMinAndRequiredStakes,
+} from '~hooks';
+import { MotionData, SetStateFn } from '~types';
+import {
+  formatStakePercentage,
+  getRemainingToStake,
+  getStakedPercentages,
+} from './helpers';
+
+export interface StakingWidgetContextValues {
+  canBeStaked: boolean;
+  canUserStakeNay: boolean;
+  totalNAYStaked: string;
+  remainingToStake: Decimal;
+  minUserStake: Decimal;
+  maxUserStake: Decimal;
+  yayPercentage: string;
+  nayPercentage: string;
+  totalPercentage: number;
+  enoughReputation: boolean;
+  enoughTokens: boolean;
+  isObjection: boolean;
+  setIsObjection: SetStateFn;
+  userActivatedTokens: Decimal;
+  isSummary: boolean;
+  setIsSummary: SetStateFn;
+  motionId: string;
+  requiredStake: Decimal;
+  userStake: Decimal;
+  nativeTokenDecimals: number;
+  nativeTokenSymbol: string;
+  reputationLoading: boolean;
+  loadingStakeData: boolean;
+  getErrorType: (limitExceeded: boolean) => string | null;
+}
+
+export const StakingWidgetContext = createContext<
+  Partial<StakingWidgetContextValues>
+>({});
+
+export const useStakingWidgetContext = () => {
+  const ctx = useContext(StakingWidgetContext);
+
+  if (!Object.keys(ctx).length) {
+    throw new Error('Could not find StakingWidgetContext');
+  }
+
+  return ctx as StakingWidgetContextValues;
+};
+
+interface StakingWidgetProviderProps {
+  children: ReactNode;
+  motionData: MotionData;
+}
+
+const StakingWidgetProvider = ({
+  children,
+  motionData: {
+    motionId,
+    motionDomainId,
+    motionStakes: { nay: totalNAYStaked },
+    motionStakes,
+    rootHash,
+    skillRep,
+  },
+}: StakingWidgetProviderProps) => {
+  const { user } = useAppContext();
+  const { colony } = useColonyContext();
+  const [isSummary, setIsSummary] = useState(false);
+  const [isObjection, setIsObjection] = useState(false);
+
+  const { data: userReputation, loading: reputationLoading } =
+    useGetUserReputationQuery({
+      variables: {
+        input: {
+          colonyAddress: colony?.colonyAddress ?? '',
+          walletAddress: user?.walletAddress ?? '',
+          domainId: Number(motionDomainId),
+          rootHash,
+        },
+      },
+    });
+
+  const { minUserStake, requiredStake, loadingStakeData } =
+    useMinAndRequiredStakes(skillRep);
+
+  // TODO
+  //   const {
+  //     data: userData,
+  //     loading: userDataLoading,
+  //   } = useUserBalanceWithLockQuery({
+  //     variables: {
+  //       address: walletAddress,
+  //       tokenAddress: nativeTokenAddress,
+  //       colonyAddress,
+  //     },
+  //   });
+  //userData?.user?.userLock?.balance || 0,
+
+  const userActivatedTokens = useMemo(
```

Comment:
These `Decimals` are memoised so they don't trigger the a rendering of the entire provider's subtree on each render